src/Cose.c

```diff
@@ -150,47 +150,59 @@ HCOSE COSE_Decode(const byte * rgbData, size_t cbData, int * ptype, COSE_object_
 	CHECK_CONDITION(cbor->type == CN_CBOR_ARRAY, COSE_ERR_INVALID_PARAMETER);
 
 	switch (*ptype) {
+#ifdef COSE_DECODE_ENCRYPT
 	case COSE_enveloped_object:
 		h = (HCOSE)_COSE_Enveloped_Init_From_Object(cbor, NULL, CBOR_CONTEXT_PARAM_COMMA perr);
 		if (h == NULL) {
 			goto errorReturn;
 		}
 		break;
+#endif
```

Comment:
I would prefer to return a new error COSE_ERR_UNSUPPORTED_COSE_TYPE rather than invalid parameter.  This would allow people to detect that it could be decoded, but the build they are using does not support it.